### PR TITLE
[PPD-78] GPD Payments

### PR DESCRIPTION
**Pull request by jacopocarlini**
### List of changes

GPD Payments:
 - app service
 - apim
 - wsdl
 - base policy

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/env/dev/terraform.tfvars [hunk 1]
```diff
@@ -26,6 +26,7 @@ cidr_subnet_buyerbanks         = ["10.1.134.0/24"]
 cidr_subnet_reporting_fdr      = ["10.1.135.0/24"]
 cidr_subnet_reporting_common   = ["10.1.136.0/24"]
 cidr_subnet_gpd                = ["10.1.138.0/24"]
+cidr_subnet_payments           = ["10.1.140.0/24"]
```
Comment (pasqualespica) [inline src/env/dev/terraform.tfvars:29]: why 140 and no 139 ?

File: src/env/dev/terraform.tfvars [hunk 2]
```diff
@@ -322,3 +323,8 @@ gpd_plan_sku_size = "S1"
 reporting_function_autoscale_minimum = 1
 reporting_function_autoscale_maximum = 3
 reporting_function_autoscale_default = 1
+
+// GPD Payments
+payments_always_on            = false
+payments_paa_id_intermediario = "77777777777"   // TODO
```
Comment (pasqualespica) [inline src/env/dev/terraform.tfvars:329]: @corvettafabio we need this configuration information in dev/uat/prod env for GDP/PAYMENT service as PT.

cc @jacopocarlini

File: src/env/prod/terraform.tfvars [hunk 3]
```diff
@@ -327,3 +328,8 @@ reporting_analysis_function_always_on         = true
 reporting_analysis_function_autoscale_minimum = 1
 reporting_analysis_function_autoscale_maximum = 3
 reporting_analysis_function_autoscale_default = 1
+
+// GPD Payments
+payments_always_on            = true
```
Comment (pasqualespica) [inline src/env/prod/terraform.tfvars:333]: set it default to `true` and only in dev setting it to `false`

File: src/env/prod/terraform.tfvars [hunk 4]
```diff
@@ -327,3 +328,8 @@ reporting_analysis_function_always_on         = true
...
+payments_paa_id_intermediario = "77777777777"   // TODO
+payments_paa_stazione_int     = "77777777777_1" // TODO
```
Comment (pasqualespica) [inline src/env/prod/terraform.tfvars:335]: ditto

File: src/env/uat/terraform.tfvars [hunk 5]
```diff
@@ -322,3 +323,9 @@ reporting_function_autoscale_default = 1
 reporting_batch_function_always_on    = true
 reporting_service_function_always_on  = true
 reporting_analysis_function_always_on = true
+
+
+// GPD Payments
+payments_always_on            = true
```
Comment (pasqualespica) [inline src/env/uat/terraform.tfvars:329]: ditto

File: src/gpd_payments.tf [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+
+
+module "payments_app_service" {
+  source = "git::https://github.com/pagopa/azurerm.git//app_service?ref=v2.2.0"
+
+  resource_group_name = azurerm_resource_group.gpd_rg.name
+  plan_type           = "external"
+  plan_id             = azurerm_app_service_plan.gpd_service_plan.id
```
Comment (pasqualespica) [inline src/gpd_payments.tf:26]: @pasqualedevita we using this `azurerm_app_service_plan.gpd_service_plan` for several things , in details : 
- for 2 app service
- and 3 fn app 
do you see possible problems in this?

**Review by pasqualespica — CHANGES_REQUESTED** [review]

File: src/env/prod/terraform.tfvars [hunk 7]
```diff
@@ -26,6 +26,7 @@ cidr_subnet_buyerbanks         = ["10.1.134.0/24"]
 cidr_subnet_reporting_fdr      = ["10.1.135.0/24"]
 cidr_subnet_reporting_common   = ["10.1.136.0/24"]
 cidr_subnet_gpd                = ["10.1.138.0/24"]
+cidr_subnet_payments           = ["10.1.140.0/24"]
```
[the remark this answers is not in the document]
Comment (jacopocarlini) [reply src/env/prod/terraform.tfvars]: fixed

**Review by pasqualespica — DISMISSED** [review]
👍

**Pull request merged**